docs/cookbook/creating-extension.rst

```diff
@@ -0,0 +1,41 @@
+How to create an extension for Sylius?
+======================================
+
+Sylius extension is nothing else but a regular Symfony bundle adding custom behaviour to the default Sylius application.
+
+The best way to create your own extension is to use `Sylius bundle skeleton <https://github.com/Sylius/BundleSkeleton>`_,
+which has built-in infrastructure for designing and testing using Behat.
+
+.. note::
+
+    This cookbook assumes that you are worked with Sylius and Symfony before.
+
+**1. Create project using Composer.**
+
+.. code-block:: bash
+
+    $ composer create-project sylius/bundle-skeleton SyliusExtensionPath
+
+**2. Get familiar with basic extension design.**
+
+The skeleton comes with simple application that greets a customer. There are features' scenarios in ``features`` directory;
+simple example bundle with a controller, a template and a routing configuration in ``src``;
+and the testing infrastructure in ``tests``.
+
+.. note::
+
+    The ``tests/Application`` directory contains a sample Symfony application built to test your extension against it.
+
+**3. Remove boilerplate files and rename your bundle.**
+
+In most cases you don't want your Sylius extension to greet customer like it is now, so feel free to remove unnecessary
+controllers, assets and feature. You will also want to change the extension namespace from ``Acme\ExampleBundle`` to a
+more meaningful one. Keep in mind that these changes also need to be done in ``tests/Application`` and ``composer.json``.
+
+**4. Implement your awesome features.**
+
+Looking at existing Sylius extensions like `Lakion\SyliusCmsBundle`_ or `Lakion\SyliusElasticSearchBundle`_ are great
```

Comment:
`` Looking at existing Sylius extensions like `Lakion\SyliusCmsBundle`_ or `Lakion\SyliusElasticSearchBundle`_ is a great way to start developing your own extensions.``